File: benchmarks/c/13_factorial.c
// Factorial Benchmark - C Implementation
// Recursive Factorial calculation

#include <stdio.h>

long long fact(int n) {
    if (n <= 1) {
        return 1;
    }
    return n * fact(n - 1);
}

int main() {
    int n = 15;
    long long result = fact(n);
    printf("fact(%d) = %lld\n", n, result);
    return 0;
}

File: benchmarks/go/13_factorial.go
// Factorial Benchmark - Go Implementation
// Recursive Factorial calculation

package main

import "fmt"

func fact(n int) int {
	if n <= 1 {
		return 1
	}
	return n * fact(n-1)
}

func main() {
	n := 15
	result := fact(n)
	fmt.Printf("fact(%d) = %d\n", n, result)
}